pages/_app.js

```diff
@@ -1,9 +1,10 @@
 import Layout from "@/components/Layout";
 import GlobalStyle from "../styles";
 import useLocalStorageState from "use-local-storage-state";
-import { plants } from "@/lib/data";
 import { uid } from "uid";
-import { categories } from "@/lib/data-categories";
+import useSWR from 'swr';
+
+const fetcher = url => fetch(url).then(response => response.json());
```

Comment:
For a proper error handling it is better to use the fetcher function that also handles error for not ok responses. [See our handout for SWR.](https://github.com/neuefische/cgn-web-23-3/blob/main/sessions/react-data-fetching/react-data-fetching.md#loading-and-error-state) and the [SWR docs](https://swr.vercel.app/docs/error-handling).

And you can think about adding a SWRConfig for declaring the fetcher function only once. :) [(see handout)](https://github.com/neuefische/cgn-web-23-3/blob/main/sessions/react-data-fetching/react-data-fetching.md#configuring-swr)